### PR TITLE
Vehicle review app

### PR DESCRIPTION
**Pull request by vkarpov15**
Now works with new stargate-mongoose and stargate JSON API. Try it out and let me know if you run into any issues!

### CODE REVIEW

File: typescript-express-reviews/.env.example [hunk 1]
```diff
@@ -0,0 +1,4 @@
+STARGATE_JSON_API_URL=http://127.0.0.1:8080/v1/reviews
```
Comment (jeffreyscarpenter) [inline typescript-express-reviews/.env.example:1]: are you supposed to source one of these environment files before running the example? The readme doesn't mention this. I got further after I did `source .env.example` before running `npm run seed`

File: typescript-express-reviews/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# typescript-express-reviews
+
+This sample demonstrates integrating Mongoose with [`express`](https://www.npmjs.com/package/express).
+
+## Setup
+
+Make sure you have a local stargate instance running as described on the [main page](../README.md) of this repo.
```
Comment (jeffreyscarpenter) [inline typescript-express-reviews/README.md:7]: also set up environment variables?

**Comment by jeffreyscarpenter** [comment]
I see that this branch is currently showing 7 commits behind main. Should we rebase? I think this PR should be tested against latest Stargate and JSONAPI. 

When I tried running this on my own desktop, I already had a docker-compose setup running from work on the jsonapi repo using newer versions, and I got the stack trace below. Everything worked after I used the docker-compose from the PR branch (`Running with Stargate v2.0.9, JSON API v1.0.0-ALPHA-2`).

```
jeffreycarpenter@jcarpenter-rmbp15 typescript-express-reviews % npm run seed

> @mongoosejs/typescript-express-reviews@1.0.0 seed
> ts-node src/seed/seed

Connecting to http://127.0.0.1:8080/v1/reviews
error: Error: Command "createCollection" failed with the following errors: [{"message":"Unrecognized field \"options\" (class io.stargate.sgv2.jsonapi.api.model.command.impl.CreateCollectionCommand), not marked as ignorable (one known property: \"name\"])\n at [Source: (ByteArrayInputStream); line: 1, column: 58] (through reference chain: io.stargate.sgv2.jsonapi.api.model.command.impl.CreateCollectionCommand[\"options\"])","exceptionClass":"UnrecognizedPropertyException"}]
    at handleIfErrorResponse (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/client/httpClient.ts:290:11)
    at HTTPClient.executeCommand (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/client/httpClient.ts:272:26)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/collections/db.ts:70:14
    at async executeOperation (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/collections/utils.ts:185:11)
    at async executeOperation (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/collections/utils.ts:185:11)
    at async Function.createCollection (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/mongoose/lib/model.js:1399:5)
    at async _createCollection (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/mongoose/lib/model.js:1315:12)
error: Error: Command "createCollection" failed with the following errors: [{"message":"Unrecognized field \"options\" (class io.stargate.sgv2.jsonapi.api.model.command.impl.CreateCollectionCommand), not marked as ignorable (one known property: \"name\"])\n at [Source: (ByteArrayInputStream); line: 1, column: 58] (through reference chain: io.stargate.sgv2.jsonapi.api.model.command.impl.CreateCollectionCommand[\"options\"])","exceptionClass":"UnrecognizedPropertyException"}]
    at handleIfErrorResponse (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/client/httpClient.ts:290:11)
    at HTTPClient.executeCommand (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/client/httpClient.ts:272:26)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/collections/db.ts:70:14
    at async executeOperation (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/collections/utils.ts:185:11)
    at async executeOperation (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/collections/utils.ts:185:11)
    at async Function.createCollection (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/mongoose/lib/model.js:1399:5)
    at async _createCollection (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/mongoose/lib/model.js:1315:12)
StargateServerError: Command "createCollection" failed with the following errors: [{"message":"Unrecognized field \"options\" (class io.stargate.sgv2.jsonapi.api.model.command.impl.CreateCollectionCommand), not marked as ignorable (one known property: \"name\"])\n at [Source: (ByteArrayInputStream); line: 1, column: 58] (through reference chain: io.stargate.sgv2.jsonapi.api.model.command.impl.CreateCollectionCommand[\"options\"])","exceptionClass":"UnrecognizedPropertyException"}]
    at handleIfErrorResponse (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/client/httpClient.ts:290:11)
    at HTTPClient.executeCommand (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/client/httpClient.ts:272:26)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/collections/db.ts:70:14
    at async executeOperation (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/collections/utils.ts:185:11)
    at async executeOperation (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/stargate-mongoose/src/collections/utils.ts:185:11)
    at async Function.createCollection (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/mongoose/lib/model.js:1399:5)
    at async _createCollection (/Users/jeffreycarpenter/stargate-mongoose-sample-apps-2/typescript-express-reviews/node_modules/mongoose/lib/model.js:1315:12) {
  errors: [
    {
      message: 'Unrecognized field "options" (class io.stargate.sgv2.jsonapi.api.model.command.impl.CreateCollectionCommand), not marked as ignorable (one known property: "name"])\n' +
        ' at [Source: (ByteArrayInputStream); line: 1, column: 58] (through reference chain: io.stargate.sgv2.jsonapi.api.model.command.impl.CreateCollectionCommand["options"])',
      exceptionClass: 'UnrecognizedPropertyException'
    }
  ],
  command: { createCollection: { name: 'authentications', options: {} } }
}
```

**Comment by jeffreyscarpenter** [comment]
Sorry for accidental close, hit wrong button.

**Comment by tatu-at-datastax** [comment]
@jeffreyscarpenter Thing wrt "options" will be resolved with Ivan's PR to allow it everywhere (we had a discussion on best ways to achieve that which slower down merging it). Should get in the next JSON API build.

File: typescript-express-reviews/README.md (hunk 2 above)
Comment (ivansenic) [reply typescript-express-reviews/README.md:7]: yes I am hitting this as well.. Can we have meaningful defaults for these, cause we anyway depend on that startup script that brings jsonapi docker compose up:

```
STARGATE_JSON_API_URL=http://localhost:8080/v1/my_keyspace
STARGATE_JSON_AUTH_URL=http://localhost:8081/v1/auth 
STARGATE_JSON_USERNAME=cassandra 
STARGATE_JSON_PASSWORD=cassandra
```

Comment (ivansenic) [reply typescript-express-reviews/README.md:7]: ah, I see the `.env` examples now, maybe just say this should be the first step `cp .env.example .env`?

**Review by ivansenic — APPROVED** [review]
Tried it out, seems to work.. Few small suggestions.. Did not get too much into the code..

File: typescript-express-reviews/.env.example (hunk 1 above)
Comment (vkarpov15) [reply typescript-express-reviews/.env.example:1]: That's a very good point. The assumption was that you would create a `.env` file from `.env.example`, but those instructions weren't in the README. Also, `.env` currently doesn't have any sensitive credentials, so we'll just commit that to GitHub.

File: typescript-express-reviews/README.md (hunk 2 above)
Comment (vkarpov15) [reply typescript-express-reviews/README.md:7]: Fixed by bd016dc28e7b5f1f3088c86bd474edb9ae8dfd33

File: typescript-express-reviews/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+## Running This Example
+
```
[the remark this answers is not in the document]
Comment (vkarpov15) [reply typescript-express-reviews/README.md:10]: Fixed in https://github.com/stargate/stargate-mongoose-sample-apps/pull/15/commits/ca2baf1e2b9ff28a921db2a66613bc2f814a8237

**Comment by vkarpov15** [comment]
Re: options, I put a PR into Mongoose to fix that. It should also be fixed in Stargate, but Mongoose 7.0.4 won't have that issue.

File: typescript-express-reviews/.env.example (hunk 1 above)
Comment (tatu-at-datastax) [reply typescript-express-reviews/.env.example]: I guess one reason to have `.env.example` (or `env.example`) is to alliw `.ignore` of real `.env` -- if everyone cloning repo is likely to modify it, avoid it being submitted as part of PR.

**Pull request merged**